### PR TITLE
Add stencil and event flags

### PR DESCRIPTION
**Pull request by apazylbe**
Adds event flags for which tests are enabled in the pipeline
Add stencil increment with a fixed colour fragment shader.

### CODE REVIEW

File: renderdoc/driver/vulkan/vk_pixelhistory.cpp [hunk 1]
```diff
@@ -525,6 +580,109 @@ struct VulkanPixelHistoryCallback : public VulkanDrawcallCallback
...
+    }
+
+    // Scissor
+    {
+      bool inRegion = false;
+      bool inAllRegions = true;
+      // Do we even need to know viewerport here?
+      const VkRect2D *pScissors;
+      uint32_t scissorCount;
+      if(p.dynamicStates[VkDynamicViewport])
```
Comment (baldurk) [inline renderdoc/driver/vulkan/vk_pixelhistory.cpp]: Is this looking at the wrong state? I'd have expected `VkDynamicScissor`.

File: renderdoc/driver/vulkan/vk_pixelhistory.cpp [hunk 2]
```diff
@@ -525,6 +580,109 @@ struct VulkanPixelHistoryCallback : public VulkanDrawcallCallback
...
+          if(p.attachments[i].blendEnable)
+          {
+            flags |= Blending_Enabled;
+            break;
+          }
+        }
+      }
+      else
+      {
+        // Might not have attachments if rasterization is disabled?
```
Comment (baldurk) [inline renderdoc/driver/vulkan/vk_pixelhistory.cpp]: I think this comment is accurate - you should add a check that `p.attachments` is non-empty.

**Review by baldurk — CHANGES_REQUESTED** [review]
Looks good overall, I noted a couple of things.

File: renderdoc/driver/vulkan/vk_pixelhistory.cpp (hunk 1 above)
Comment (apazylbe) [reply renderdoc/driver/vulkan/vk_pixelhistory.cpp]: You are right, thanks!

File: renderdoc/driver/vulkan/vk_pixelhistory.cpp (hunk 2 above)
Comment (apazylbe) [reply renderdoc/driver/vulkan/vk_pixelhistory.cpp]: Added.

**Review by baldurk — APPROVED** [review]

**Pull request merged**